acme/acme/__init__.py

```diff
@@ -10,3 +10,17 @@
   https://github.com/ietf-wg-acme/acme/tree/draft-ietf-acme-acme-01
 
 """
+import sys
+
+# This code exists to keep backwards compatibility with people using acme.jose
+# before it became the standalone josepy package.
+#
+# It is based on
+# https://github.com/requests/requests/blob/1278ecdf71a312dc2268f3bfc0aabfab3c006dcf/requests/packages.py
+
+jose = __import__('josepy')
+for mod in list(sys.modules):
```

Comment:
That's a good point.

I'm worried a comment in `__init__.py` might get missed when adding new modules to josepy because the author of those modules might not need to look in that file.

I'm tempted to add code that tries to recursively walk `josepy` and either import everything or test that everything that's not a test file has been imported for `acme.jose`.

Alternatively, I'm tempted to see if we can use the `warning` module to tell people to stop using `acme.jose` and switch to `josepy`.

I have no idea how to do either of those things off the top of my head.

Reply:
My first choice would be to decide that backward-compat renaming for new submodules is unnecessary, since new code should use the new package name. My second is to cordon the ugly FS-walking code (remember to use pkg_resources instead) away in a test rather than glumping up import time with strange, failure-prone things that could even equate to an arbitrary eval if directory privs were set wrong. :-)